add docs on alpha- and betanets (#1264)
ethereum-optimism/docs: pages/stack/public-devnets.mdx

---

> title: Public Devnets 
> lang: en-US
> description: An overview of the devnets (alphanets and betanets) which can be used to test upcoming OP Stack releases. 

Reviewer: ```suggestion
title: Public devnets 
lang: en-US
description: Learn about alphanets and betanets - the development networks for testing upcoming OP Stack releases. 
```

---

import { Callout } from 'nextra/components'

> OP Labs maintains public devnets, which are designed to streamline testing, iteration, and deployment of new features for the OP Stack.

Reviewer: ```suggestion
#Overview

OP Labs maintains development networks (devnets) to test and deploy new OP Stack features. These networks come in two types: alphanets and betanets.
```

> There are two different kinds of public devnets: *alphanets* and *betanets*, which serve different purposes and are deployed on different timelines. 
> 

Reviewer: ```suggestion

```

> # Alphanets 
> 
> **Alphanets** are monthly devnets that include production-bound protocol upgrades that are ready for testing. 
> We rely on alphanets to move faster while improving the reliability of upgrades. 
> If a feature is ready by the monthly deployment date (typically the third Wednesday of the month), the feature will be included in that month's devnet. 
> Otherwise, the feature must wait for the next alphanet. 
> Features _must_ be included on an alphanet before they can be deployed on a betanet.
> 
> Alphanets decouple feature development and testing from hardfork scheduling, allowing us to test and iterate on features well in advance of their release on mainnet.
> 
> They're designed to stay fresh: Each one is deployed monthly and replaced, so they won't be useable as long-term networks. 
> Each alphanet is launched from genesis, meaning that new features are encoded in genesis rather than undergoing a network upgrade from a preexisting version. 
> The alphanet is then tested, with an emphasis on acceptance testing for new features followed by fault injection testing and performance / load testing. 
> After five weeks, the alphanet is spun down and a new alphanet takes its place, with whatever new features have been developed in that time. 
> 
> Alphanets are named after animals and ordered alphabetically.

Reviewer: ```suggestion
## Network types

### Alphanets

Alphanets are monthly networks that test production-ready protocol upgrades.
They enable faster iteration while improving upgrade reliability.

Key characteristics:

*   Deploy on the third Wednesday of each month
*   Include features ready by deployment date
*   Start fresh from genesis with each deployment
*   Run for five weeks
*   Follow an alphabetical naming convention based on animal names
*   Must precede betanet deployment

Testing phases:

1.  Feature acceptance testing
2.  Fault injection testing
3.  Performance and load testing
```

Author: Hi @krofax, thanks for taking a look at this and leaving suggested edits! 

Can you give me a little context on the suggested change here? Is it better to have this kind of content formatted in a list? Was my original text too long? I’m happy to rework this but I think the new version also needs some updates for accuracy, so I want to make sure that I’m keeping the changes consistent with what you’re looking for.

Reviewer: Hi @tessr,
Thanks for being open to suggestions. 
Let me explain the reasoning behind the restructuring:

1. Improved scannability:

- Breaking down the content into bullet points and shorter sections makes it easier for developers to quickly find information.
- The lists help readers grasp key points without having to parse through paragraphs
- The hierarchical structure (using ###) creates clear visual separation between sections

2. Content organization:

- Grouped related information under clear subheadings
- Moved from narrative style to more technical documentation format
- Maintained all the important technical details while making them more accessible

3. Technical accuracy:

- You're absolutely right about accuracy updates being needed! The core information remains the same, but if there are specific technical details that need to be preserved or clarified, we should definitely incorporate those
- We can combine your detailed technical content with this more structured format

Given our timeline, I'm happy to approve your current version so we can get this out quickly. Perhaps we could collaborate on implementing these structural improvements in a future update? Let me know if you'd be interested in working together on that.

Author: OK, great, thank you! This reasoning is very clear. 

If it's OK with you, I'll go ahead and merge this today more or less as-is, and then we can iterate on it. What might be easiest is if you open another PR to reformat it and then I can drop in the small accuracy fixes.

> # Betanets 
> 
> **Betanets** are devnets that are deployed on an as-needed basis, typically ahead of an upcoming hardfork. 
> Features are activated using hardfork timestamps, mimicking the production upgrade process.
> 
> Betanets are used to solidify the scope of the next hardfork before deploying to the production testnet (e.g., OP Sepolia). 
> A betanet includes the exact set of features that we expect to include in the next network upgrade.
> Features _must_ be included on a betanet before they can be deployed on a testnet.

Reviewer: ```suggestion
### Betanets

Betanets test specific upcoming hardforks. They mimic the production upgrade process with features activated using hardfork timestamps.

Key characteristics:

*   Deploy as needed before hardforks
*   Activate features using hardfork timestamps
*   Match expected production upgrade process
*   Run until next betanet deployment
*   Must precede testnet deployment

```

> 
> # Devnet comparison
> 
> | Network type        | Alphanet                                   | Betanet                                   |
> | ------------------- | -------------------------------------------| ----------------------------------------- |
> | Purpose             | Early testing of production-bound features | Final validation for the next hardfork    |
> | Deployment          | Monthly                                    | As-needed                                 |
> | Feature activation  | Included and activated at genesis          | Activated through a network upgrade       |
> | Lifecycle           | 5 weeks                                    | Until replaced by the next betanet        |
> 

Reviewer: ```suggestion

### Devnet comparison

| Attributes | Alphanet          | Betanet             |
| ---------- | ----------------- | ------------------- |
| Purpose    | Test new features | Validate hardforks  |
| Deployment | Monthly           | As needed           |
| Activation | At genesis        | Via network upgrade |
| Duration   | 5 weeks           | Until next betanet  |

```

# Current devnets 

> 
> **Alpaca** 
> * Launch date: 2025-01-15
> * Anticipated end-of-life: 2025-02-19
> * Scope: Adds updates to the L1 contracts to support upgrades via OP Contracts Manager.
> * RPC endpoint: https://alpaca-0.optimism.io
> * `manifest.yaml`: https://github.com/ethereum-optimism/devnets/blob/main/alphanets/alpaca/manifest.yaml
> 

Reviewer: ```suggestion

**Alpaca** 
*   Start date: 2025-01-15
*   End date: 2025-02-19
*   Scope: Updates L1 contracts to support upgrades via OP Contracts Manager
*   Access:
    *   RPC endpoint: [https://alpaca-0.optimism.io](https://alpaca-0.optimism.io)
    *   Config: [https://github.com/ethereum-optimism/devnets/blob/main/alphanets/alpaca/manifest.yaml](https://github.com/ethereum-optimism/devnets/blob/main/alphanets/alpaca/manifest.yaml)

```
